processor/transformer/transformer.go

```diff
@@ -375,16 +354,68 @@ func (trans *HandleT) request(ctx context.Context, url string, data []Transforme
 			trans.logger.Errorf("Transformer returned : %v", string(respData))
 			respData = []byte(fmt.Sprintf("Failed to unmarshal transformer response: %s", string(respData)))
 			transformerResponses = nil
-			resp.StatusCode = 400
+			statusCode = 400
 		}
 	}
 
-	if resp.StatusCode != http.StatusOK {
+	if statusCode != http.StatusOK {
 		for i := range data {
 			transformEvent := &data[i]
-			resp := TransformerResponseT{StatusCode: resp.StatusCode, Error: string(respData), Metadata: transformEvent.Metadata}
+			resp := TransformerResponseT{StatusCode: statusCode, Error: string(respData), Metadata: transformEvent.Metadata}
 			transformerResponses = append(transformerResponses, resp)
 		}
 	}
 	return transformerResponses
 }
+
+func (trans *HandleT) doPost(ctx context.Context, rawJSON []byte, url string, tags stats.Tags) ([]byte, int) {
+	var (
+		retryCount int
+		resp       *http.Response
+		respData   []byte
+	)
+	err := backoff.RetryNotify(
+		func() error {
+			var reqErr error
+			s := time.Now()
+			trace.WithRegion(ctx, "request/post", func() {
+				resp, reqErr = trans.Client.Post(url, "application/json; charset=utf-8", bytes.NewBuffer(rawJSON))
+			})
+			trans.requestTime(tags, time.Since(s))
+
+			if reqErr == nil {
+				// If no err returned by client.Post, reading body.
+				// If reading body fails, retrying.
+				respData, reqErr = io.ReadAll(resp.Body)
+				resp.Body.Close()
+			}
+			if reqErr != nil {
+				return reqErr
+			}
```

Comment:
I find this way a bit [more readable](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88):

```suggestion
                        
			if reqErr != nil {
				return reqErr
			}
			defer resp.Body.Close()

			// If no err returned by client.Post, reading body.
			// If reading body fails, retrying.
			respData, reqErr = io.ReadAll(resp.Body)
							
```

Reply:
Will require a repetition of the following at the end:
````
if reqErr != nil {
	return reqErr
}
```